dashboard/test/models/blockly_test.rb

```diff
@@ -142,6 +143,169 @@ def assert_equal_xml(a, b)
     assert_equal xml, xml2
   end
 
+  test 'localized shared_blocks' do
+    test_locale = :"te-ST"
+    I18n.locale = test_locale
+    custom_i18n = {
+      "data" => {
+        "blocks" => {
+          "DanceLab_atSelectColor" => {
+            "text" => "kat {TIMESTAMP} {COLOR}",
+            "options" => {
+              "COLOR" => {
+                "first": "red",
+                "second": "blue",
+              }
+            }
+          }
+        }
+      }
+    }
+
+    I18n.backend.store_translations test_locale, custom_i18n
+
+    level = create(:level, :blockly, level_num: 'level1_2_3')
+
+    custom_block =
+      [{
+        name: "DanceLab_atSelectColor",
+        pool: "SelectColor",
+        category: "Events",
+        config:
+        {
+          "color" => [140, 1, 0.74],
+          "func" => "atSelectColor",
+          "blockText" => "cat {TIMESTAMP} {COLOR}",
+          "args" => [
+            {"name" => "TIMESTAMP", "type" => "Number", "field" => true},
+            {"name" => "COLOR", "options" => [["red", "\"red\""], ["blue", "\"blue\""]]}
+          ],
+          "eventBlock" => true
+        },
+        helperCode: nil
+      }]
+
+    translated_block =
+      [{
+        name: "DanceLab_atSelectColor",
+        pool: "SelectColor",
+        category: "Events",
+        config:
+          {
+            "color" => [140, 1, 0.74],
+            "func" => "atSelectColor",
+            "blockText" => "kat {TIMESTAMP} {COLOR}",
+            "args" => [
+              {"name" => "TIMESTAMP", "type" => "Number", "field" => true},
+              {"name" => "COLOR", "options" => [["red", "\"red\""], ["blue", "\"blue\""]]}
+            ],
+            "eventBlock" => true
+          },
+        helperCode: nil
+      }]
+
+    custom_block_copy = custom_block
+    localized_custom_blocks = level.localized_shared_blocks(custom_block_copy)
+
+    assert_equal localized_custom_blocks, translated_block
+    assert_not_equal localized_custom_blocks, custom_block
+  end
+
+  test 'nil is returned if level_objects is blank' do
+    test_locale = :"te-ST"
+    I18n.locale = test_locale
+    custom_i18n = {
+      "data" => {
+        "blocks" => {
+          "DanceLab_atSelectColor" => {
+            "text" => "kat {TIMESTAMP} {COLOR}",
+            "options" => {
+              "COLOR" => {
+                "first": "red",
+                "second": "blue",
+              }
+            }
+          }
+        }
+      }
+    }
+
+    I18n.backend.store_translations test_locale, custom_i18n
+
+    level = create(:level, :blockly, level_num: 'level1_2_3')
+
+    custom_block = []
+
+    custom_block_copy = custom_block
+    localized_custom_blocks = level.localized_shared_blocks(custom_block_copy)
+
+    assert_nil localized_custom_blocks
+  end
+
+  test 'does not return a translated string if block text is not present' do
```

Comment:
seems like this test would be a lot more useful if either the i18n or the block were missing the text, but it's not particularly useful as-is, when both of them are.

It would also be valuable to see a test for the situation when we do not have any i18n information for the block being translated!